### PR TITLE
Refactor BasicCParser::unescapeString()

### PR DESCRIPTION
**Pull request by snumlautoken**
BasicCParser::unescapeString() is currently one of the methods with the highest cyclometic complexities in the entire project. The purpose of the refactoring is to reduce complexity and to add consistency in regards to the switch-case.

### CODE REVIEW

File: src/main/java/org/apache/commons/imaging/common/BasicCParser.java [hunk 1]
```diff
@@ -417,4 +341,103 @@ public static void unescapeString(final StringBuilder stringBuilder, final Strin
                     + "unterminated escape sequence found in string");
         }
     }
+
+    private static int appendHex(int i, final StringBuilder stringBuilder, final String string)
```
Comment (kinow) [inline src/main/java/org/apache/commons/imaging/common/BasicCParser.java:345]: Could you add some javadoc to these methods too, please, @snumlautoken ? The names are self-explanatory, but might still be helpful to be explicit about it, and also mention the use of the parameters.

**Review by kinow — COMMENTED** [review]
Hi @snumlautoken , thanks a lot for your PR.

The refactor seems to create a good separation of concerns with these new private methods :+1: 

I've enabled the GH Action workflow and left one comment. Let's wait for the workflow to finish to see the coverage (should ideally be the same or higher, although it could vary due to new lines for method names, etc).

Cheers
-Bruno

File: src/main/java/org/apache/commons/imaging/common/BasicCParser.java (hunk 1 above)
Comment (snumlautoken) [reply src/main/java/org/apache/commons/imaging/common/BasicCParser.java:353]: This has been added!

**Comment by kinow** [comment]
Merged, thanks!

**Pull request closed without merging**